### PR TITLE
Convert to CSV/Excel options

### PR DESCRIPTION
**Pull request by romifz**
Ref #26 #85

I took @Cristobal961's collaboration and added a tree selection input using [jsTree](https://www.jstree.com/). From all options I had a look at I think this one is the best, but any feedback is appreciated. 

screenshot

### CODE REVIEW

File: locale/es/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -518,8 +568,35 @@ msgstr ""
...
+#: default/templates/default/snippets/unflatten-options-modal.html:7
+#: default/templates/default/to-spreadsheet.html:106
+msgid "Options"
+msgstr "Opciones"
+
+#: default/templates/default/snippets/unflatten-options-modal.html:30
+#, fuzzy
+#| msgid "Please choose an operation type"
+msgid "Please choose at least one output format."
+msgstr "Elija un tipo de operación"
```
Comment (aguilerapy) [inline locale/es/LC_MESSAGES/django.po:580]: Aquí creo que el punto al final del mensaje en Inglés se puede borrar, y la traducción debería ser algo como "Favor elija al menos un tipo de operación".

File: tests/fixtures/1.1/release-packages/ocds-213czf-000-00001.json [hunk 2]
```diff
@@ -0,0 +1,1143 @@
+{
```
Comment (aguilerapy) [inline tests/fixtures/1.1/release-packages/ocds-213czf-000-00001.json:1]: The name of this file should not be "0004-package" or something like that?

File: tests/test_get_schema_options.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from django.test import TestCase
+
+
+class TestSchemaOptions(TestCase):
+
+    def test_get_options(self):
+        pass
```
Comment (aguilerapy) [inline tests/test_get_schema_options.py:7]: This is necessary/correct?

**Comment by aguilerapy** [comment]
Great! I have some questions.

In "Options" popup, we should not explain that "Filter data by field" it's optional? Or does the non-bold font tell us?

After the error message "Define both the field and value to filter data", scroll ball obeys the main toucan page, not the "Options" popup, so it's not possible to close the window without resizing the page.

Also, it is not validated that the entered text is correct, such as "tenderr" in the filter.

imagen

**Comment by romifz** [comment]
> In "Options" popup, we should not explain that "Filter data by field" it's optional? Or does the non-bold font tell us?

All options are optional except for Output formats (which should show a warning if there is not at least one selected option), we can add an asterisk to it, but I think it's kind of obvious that you shouldn't add a filter value if you don't want to filter data. The warning on "Filter data by field" is shown because the filter field and filter value fields should be used together.
 
> After the error message "Define both the field and value to filter data", scroll ball obeys the main toucan page, not the "Options" popup, so it's not possible to close the window without resizing the page.

I'll add another close button at the header. The popup can be closed by clicking outside of it too.

> Also, it is not validated that the entered text is correct, such as "tenderr" in the filter.

We could have a dynamic select box there, so the user doesn't have to write.

File: locale/es/LC_MESSAGES/django.po (hunk 1 above)
Comment (romifz) [reply locale/es/LC_MESSAGES/django.po:578]: Cambio eso

File: tests/fixtures/1.1/release-packages/ocds-213czf-000-00001.json (hunk 2 above)
Comment (romifz) [reply tests/fixtures/1.1/release-packages/ocds-213czf-000-00001.json:1]: I don't think so

File: tests/test_get_schema_options.py (hunk 3 above)
Comment (romifz) [reply tests/test_get_schema_options.py]: No, I didn't use this test, I'm removing it.

**Comment by jpmckinney** [comment]
Are we getting close to completing this PR? If necessary, we should reduce the scope so that we can get it merged, and then continue in new issues.

**Review by aguilerapy — APPROVED** [review]

**Comment by jpmckinney** [comment]
🎉 @romifz Are you deploying it? I'm curious to see it in production, and share it with Sabah.

**Comment by yolile** [comment]
@jpmckinney deployed!

**Pull request merged**